cpp/client/ct.cc

```diff
@@ -753,7 +755,7 @@ static void DiagnoseCertChain() {
       continue;
     }
     LOG(INFO) << "SCT number " << i + 1 << ":\n" << sct.DebugString();
-    if (verifier != NULL) {
+    if (verifier.get()) {
```

Comment:
On Thu, 20 Aug 2015 at 15:38 Pierre Phaneuf <notifications@github.com>
wrote:

> In cpp/client/ct.cc
> <https://github.com/google/certificate-transparency/pull/874#discussion_r37536309>
> :
>
> > @@ -753,7 +755,7 @@ static void DiagnoseCertChain() {
> >        continue;
> >      }
> >      LOG(INFO) << "SCT number " << i + 1 << ":\n" << sct.DebugString();
> > -    if (verifier != NULL) {
> > +    if (verifier.get()) {
>
> No, @benlaurie <https://github.com/benlaurie> is just remembering "C from
> the late 70's", due to the CDC 6000 he keeps around to work on OpenSSL... [image:
> :wink:]
>

Fun fact: second machine I ever coded for and first assembler I learnt was,
in fact, a CDC 6400.

This was explicitly legislated as "proper C++" a long time ago (and proper
> C as well, from at least C89, I think?):
>
>
> https://github.com/cplusplus/draft/blob/093c1eebd3e2a1682331e8b1bd16677cbdb79d6f/source/conversions.tex#L549
>

Actually, its not to do with legality, its to do with:

a) preference,

b) static analysis (one thing I've been playing with recently is ensuring
that non-bool values are correctly tested).

—
> Reply to this email directly or view it on GitHub
> <https://github.com/google/certificate-transparency/pull/874/files#r37536309>
> .
>